### PR TITLE
Fix/issue 2698 - Include street address when recalculating the tax in edit order page.

### LINKED ISSUE #2698: Incorrect tax returned for manually created orders
**Describe the bug**
When using automated tax rates from WooCommerce Shipping and Tax, incorrect tax rates are being returned for manually created orders. However, the correct tax rate is returned for orders created on the front end.

**Steps to replicate**
1. Install [WooCommerce Shipping and Tax](https://wordpress.org/plugins/woocommerce-services/) and enable automated tax calculations in WooCommerce → Settings → Tax. Make sure your store's address is in WA.
2. Go to your shop page, add items to cart and on the checkout use the shipping address `1141 Cays Rd., Sequim, WA 98382`
3. Notice that you get a tax rate of 8.6%, which is the correct/expected tax as per TaxJar

https://github.com/Automattic/woocommerce-services/assets/11092629/e70a179e-74bf-4039-a8f3-d96a44675c57

4. Now go to WooCommerce → Orders and manually create an order. Use the same shipping address `1141 Cays Rd., Sequim, WA 98382` and click the recalculate button for taxes to be pulled.
5. Notice that you get a tax rate of 8.9%, which is incorrect.

https://github.com/Automattic/woocommerce-services/assets/11092629/75d84ff8-2006-4775-8f98-ed209862d50e

**Expected behavior**
The automatic tax rate returned in manual orders should correspond to the tax rate given on the front end for the same address.
It seems WooCommerce Shipping & Tax is picking an incorrect rate.

Here are logs from the test:
[WCS logs from testing.txt](https://github.com/Automattic/woocommerce-services/files/13412649/WCS.logs.from.testing.txt)

Discussed with devs here - p1700145231014539-slack-C7U3Y3VMY.
Ticket: 7275721-zd-a8c

### PR DESCRIPTION
**Pull request by iyut**
## Description
This PR fix the incorrect tax rate when the user try to recalculate the tax from edit order page. It fix the tax rate by enqueueing the **new-order-taxjar.js** file on both new and edit order page.

### Related issue(s)
Fixes #2698 

### Steps to reproduce & screenshots/GIFs
1. Install [WooCommerce Shipping and Tax](https://wordpress.org/plugins/woocommerce-services/) and enable automated tax calculations in WooCommerce → Settings → Tax. Make sure your store's address is in WA.
2. Go to your shop page, add items to cart and on the checkout use the shipping address `1141 Cays Rd., Sequim, WA 98382`
3. Notice that you get a tax rate of 8.6%, which is the correct/expected tax as per TaxJar

https://github.com/Automattic/woocommerce-services/assets/11092629/e70a179e-74bf-4039-a8f3-d96a44675c57

4. Now go to WooCommerce → Orders and manually create an order. Use the same shipping address `1141 Cays Rd., Sequim, WA 98382` and click the recalculate button for taxes to be pulled.
5. Notice that you get a tax rate of 8.6%, which is correct.
6. Be sure to test with HPOS enabled and disabled.

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added
- [x] `readme.txt` entry added

### CODE REVIEW

**Review by bartech — APPROVED** [review]
Looks good!!! 🚀 
Now script is enqueued and tax rate is calculated correctly on Order Edit.

**Review by bartech — COMMENTED** [review]
`wc-taxjar-order` script is loaded on Order Edit screen on Legacy and HPOS.
One note though script is also loaded on Order List screen.

File: readme.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ Tags: shipping, stamps, usps, woocommerce, taxes, payment, dhl, labels
 Requires PHP: 5.6
 Requires at least: 4.6
```
Comment (dustinparker) [inline readme.txt]:
```suggestion
Requires at least: 6.2
```

File: readme.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ Tags: shipping, stamps, usps, woocommerce, taxes, payment, dhl, labels
 Requires PHP: 5.6
```
Comment (dustinparker) [inline readme.txt]:
```suggestion
Requires PHP: 7.3
```

File: woocommerce-services.php [hunk 3]
```diff
@@ -10,7 +10,7 @@
  * Version: 2.4.0
  * Requires at least: 4.6
```
Comment (dustinparker) [inline woocommerce-services.php]:
```suggestion
 * Requires at least: 6.2
```

File: readme.txt [hunk 4]
```diff
@@ -4,7 +4,7 @@ Tags: shipping, stamps, usps, woocommerce, taxes, payment, dhl, labels
...
 Requires at least: 4.6
 Tested up to: 6.3
-WC requires at least: 3.6
+WC requires at least: 8.0
 WC tested up to: 8.2
```
Comment (dustinparker) [inline readme.txt]:
```suggestion
WC tested up to: 8.3
```

File: woocommerce-services.php [hunk 5]
```diff
@@ -10,7 +10,7 @@
...
  * Tested up to: 6.3
- * WC requires at least: 3.6
+ * WC requires at least: 8.0
  * WC tested up to: 8.2
```
Comment (dustinparker) [inline woocommerce-services.php]:
```suggestion
 * WC tested up to: 8.3
```

File: woocommerce-services.php [hunk 6]
```diff
@@ -10,7 +10,7 @@
  * Version: 2.4.0
  * Requires at least: 4.6
  * Tested up to: 6.3
```
Comment (dustinparker) [inline woocommerce-services.php]:
```suggestion
 * Tested up to: 6.4
```

File: readme.txt [hunk 7]
```diff
@@ -4,7 +4,7 @@ Tags: shipping, stamps, usps, woocommerce, taxes, payment, dhl, labels
 Requires PHP: 5.6
 Requires at least: 4.6
 Tested up to: 6.3
```
Comment (dustinparker) [inline readme.txt]:
```suggestion
Tested up to: 6.4
```

**Review by dustinparker — APPROVED** [review]
Just a few version updates and this is good to go

**Review by bartech — APPROVED** [review]
👍 script is now loaded only on Order Edit screen both on Legacy and HPOS.

**Pull request merged**